### PR TITLE
Envio Greeter Contract Indexing Tutorial

### PR DESCRIPTION
**Pull request by SvenMuller95**

### CODE REVIEW

File: blog/2023-09-12-index-live-greeter-contract-using-envio.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+authors:
+  - name: Sven Muller
+    url: https://twitter.com/svenmuller95
+hide_table_of_contents: false
+date: 2023-09-12
+---
+
+Blockchains excel at decentralizing data and continue to revolutionize traditional systems of trust and record-keeping, but data retrieval is often inefficient and majorly time-consuming for developers hampering the overall development experience.
+
+Indexer like Envio allow developers to sift through blockchain data, tidy it up, and store it in a useful intermediary format that is easily accessible. Applications can then use this format for potent queries on indexed data, ensuring scalability and production readiness.
```
Comment (DenhamPreen) [inline blog/2023-09-12-index-live-greeter-contract-using-envio.md]:
```suggestion
Indexers like Envio allow developers to sift through blockchain data, tidy it up, and store it in a useful intermediary format that is easily accessible. Applications can then use this format for potent queries on indexed data, ensuring scalability and production readiness.
```

File: blog/2023-09-12-index-live-greeter-contract-using-envio.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+This tutorial will take you through a step by step guide to indexing a Greeter smart contract deployed on Linea using Envio.  
+
+## Background
+
+### [Greeter contract](https://github.com/Float-Capital/hardhat-template)
```
Comment (DenhamPreen) [inline blog/2023-09-12-index-live-greeter-contract-using-envio.md]: Can you change this url to the verified contract (on the code tab) on linea scan build

File: blog/2023-09-12-index-live-greeter-contract-using-envio.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+  "Typescript"
+  "Rescript"
+[↑↓ to move, enter to select, type to filter]
+```
+
+### Start the indexer
+
+> Dev note: 📢 make sure you have docker open
+
+The following commands will start the docker and create databases for indexed data, make sure to re-run `dev` if you make changes to the files
```
Comment (DenhamPreen) [inline blog/2023-09-12-index-live-greeter-contract-using-envio.md]:
```suggestion
The following commands will start the docker and create databases for indexing data, make sure to re-run `dev` if you make changes to the files
```

File: blog/2023-09-12-index-live-greeter-contract-using-envio.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+Run
+```bash
+envio dev
+```
+
+The indexer will then start indexing the contract/s specified in the `config.yaml` file from the `start_block` specified.
```
Comment (DenhamPreen) [inline blog/2023-09-12-index-live-greeter-contract-using-envio.md]:
```suggestion
The indexer will then start indexing the contract's specified in the `config.yaml` file from the `start_block` specified.
```

File: blog/2023-09-12-index-live-greeter-contract-using-envio.md [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+### Write to contract on Lineascan
```
Comment (DenhamPreen) [inline blog/2023-09-12-index-live-greeter-contract-using-envio.md]:
```suggestion
### Write to the contract on Lineascan
```

File: blog/2023-09-12-index-live-greeter-contract-using-envio.md [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+
+Once the indexer is running, you can call functions on the Greeter contract that is deployed on Linea.
+
+Navigate to the contract on [Lineascan](https://lineascan.build/address/0xdEe21B97AB77a16B4b236F952e586cf8408CF32A#writeContract) and call the `setGreeting` function.
+
+
+### View the indexed results
+
+You can view the indexed results on a local Hasura server.
```
Comment (DenhamPreen) [inline blog/2023-09-12-index-live-greeter-contract-using-envio.md]:
```suggestion
You can view the indexed results on your local Hasura server.
```

File: blog/2023-09-12-index-live-greeter-contract-using-envio.md [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+
+Blockchains excel at decentralizing data and continue to revolutionize traditional systems of trust and record-keeping, but data retrieval is often inefficient and majorly time-consuming for developers hampering the overall development experience.
+
+Indexers like Envio allow developers to sift through blockchain data, tidy it up, and store it in a useful intermediary format that is easily accessible. Applications can then use this format for potent queries on indexed data, ensuring scalability and production readiness.
+
+This tutorial will take you through a step by step guide to indexing a Greeter smart contract deployed on Linea using Envio.  
+
+## Background
+
+### [Greeter contract](https://github.com/Float-Capital/hardhat-template)
```
Comment (SvenMuller95) [inline blog/2023-09-12-index-live-greeter-contract-using-envio.md]:
```suggestion
### [Greeter contract](https://lineascan.build/address/0xdEe21B97AB77a16B4b236F952e586cf8408CF32A#code)
```

**Review by DenhamPreen — APPROVED** [review]

**Pull request merged**